### PR TITLE
[RW-5148][risk=no] Fix user creation via directory service

### PR DESCRIPTION
**Pull request by gjuggler**
This was a simple regression due to some refactoring I'd done in https://github.com/all-of-us/workbench/pull/3708. It turns out our Profile API model uses 'username' with a different meaning than our codebase, which I overlooked.

---
**PR checklist**

- [X] This PR meets the Acceptance Criteria in the JIRA story
- [X] The JIRA story has been moved to Dev Review
- [X] This PR includes appropriate unit tests

### CODE REVIEW

File: api/src/main/resources/workbench-api.yaml [hunk 1]
```diff
@@ -5024,7 +5024,9 @@ definitions:
         type: integer
         format: int64
       username:
-        description: researchallofus username
+        description: 'The username prefix, e.g. "gjordan". Note that this is different
+          from the use of "username" elsewhere in the RW codebase, where username represents
+          the full email address.'
```
Comment (jaycarlton) [inline api/src/main/resources/workbench-api.yaml:5029]: Can you make it "full GSuite email address" so it's more foolproof?

File: api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java [hunk 2]
```diff
@@ -120,14 +120,15 @@
   private static final String CURRENT_POSITION = "Tester";
   private static final String FAMILY_NAME = "Bobberson";
   private static final String GIVEN_NAME = "Bob";
+  private static final String GSUITE_DOMAIN = "Bob";
```
Comment (jaycarlton) [inline api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java:123]: To make the errors or debug info clearer, I'd make this look more like a domain, e.g. "bobby.com".

Comment (gjuggler) [reply api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java:123]: Yep, that was copy-pasta mistake, done.

File: api/src/main/resources/workbench-api.yaml (hunk 1 above)
Comment (gjuggler) [reply api/src/main/resources/workbench-api.yaml:5029]: Done

**Comment by gjuggler** [comment]
e2e failure is unrelated, fixed by https://github.com/all-of-us/workbench/pull/3732.

**Pull request merged**